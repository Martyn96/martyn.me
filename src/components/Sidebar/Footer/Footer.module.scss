@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.footer {
  color: lighten($color-gray, 18%);
  font-size: $typographic-small-font-size;
}
