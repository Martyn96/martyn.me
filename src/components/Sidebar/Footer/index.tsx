import React from 'react';
import { OutboundLink } from 'gatsby-plugin-google-gtag';
import styles from './Footer.module.scss';

type Props = {
  copyright: string;
  repo: string;
};

export const Footer = ({ copyright, repo }: Props): JSX.Element => (
  <div className={styles.footer}>
    Source @{' '}
    <OutboundLink href={repo} rel='noopener noreferrer' target='_blank'>
      GitHub
    </OutboundLink>
    <br />
    {copyright}
  </div>
);
