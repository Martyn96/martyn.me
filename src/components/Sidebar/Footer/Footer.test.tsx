import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '.';

describe('Footer', () => {
  it('renders correctly', () => {
    const props = {
      copyright: 'copyright',
      repo: 'github',
    };

    const tree = renderer.create(<Footer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
