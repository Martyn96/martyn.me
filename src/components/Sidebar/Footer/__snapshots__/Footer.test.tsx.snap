// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders correctly 1`] = `
<div
  className="footer"
>
  Source @
   
  <a
    href="github"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    GitHub
  </a>
  <br />
  copyright
</div>
`;
